tests/eval/test_2fa_duo.py

```diff
@@ -0,0 +1,408 @@
+                found_rememberme = True
+            else:
+                assert cookie['name'] not in wanted_cookie_names
+        assert found_rememberme is True
+
+        # c Navigate to the Eval "Token authn" link on (https://diafine12.sandbox.iam.s.uw.edu/shibevalmfa)
+        sp = ServiceProviderInstance.diafine12
+        with utils.using_test_sp(sp):
+            fresh_browser.get(f'{sp_url(sp)}/shibevalmfa')
+            password = secrets.test_accounts.password
+            fresh_browser.send_inputs(netid3, password)
+            fresh_browser.click(Locators.submit_button)
+            fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        fresh_browser.close()
+
+
+def test_forget_me_admin():
+    """
+    2FA-10 Forget me (admin) per Michael, this test is not yet ready to be automated.
+    """
+
+
+def test_forget_me_self_service(utils, sp_url, sp_domain, secrets, netid3, new_tab):
+    """
+    2FA-11 Forget me (self-service)
+    """
+    # diafine6, sptest03
+
+    fresh_browser = Chrome()
+    sp = ServiceProviderInstance.diafine6
+    with utils.using_test_sp(sp):
+        fresh_browser.get(f'{sp_url(sp)}/shibevalmfa')
+        password = secrets.test_accounts.password
+        fresh_browser.send_inputs(netid3, password)
+        fresh_browser.click(Locators.submit_button)
+        fresh_browser.wait_for_tag('p', 'Use your 2FA device.')
+        fresh_browser.find_element_by_name('rememberme').click()
+        fresh_browser.switch_to.frame(fresh_browser.find_element_by_id('duo_iframe'))
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        passcode = secrets.test_accounts.duo_code
+        fresh_browser.execute_script("document.getElementsByClassName('passcode-input')[0].value=arguments[0]",
+                                     passcode)
+        fresh_browser.snap()
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        # go to an idp site to retrieve the rememberme cookie
+        fresh_browser.get('https://idp-eval.u.washington.edu/idp')
+        fresh_browser.wait_for_tag('h1', 'not found')
+        # look for this cookie:
+        # uw-rememberme-sptest03
+        cookie_list = fresh_browser.get_cookies()
+        wanted_cookie_name = 'uw-rememberme-sptest03'
+
+        found_cookie = False
+        cookie_removed = True
+        for cookie in cookie_list:
+            # print('cookie.values ', cookie.values())
+            if wanted_cookie_name in cookie.values():
+                found_cookie = True
+
+        # b. Navigate to https://idp.u.washington.edu/forgetme (https://idp-eval.u.washington.edu/forgetme)?
+        # IdP responds with "The 2FA "Remember me on this browser" setting has been removed from this browser."
+        # The cookie named uw-rememberme-sptest03 should be gone.
+        # Close browser.
```

Comment:
Please remove this. This exactly repeats what is in the code block below, except that it's already wrong because you aren't closing the browser here. :)

Reply:
Ha, I think I had the placeholder in place before I wrote the test, then I wrote the test and forgot to take out the placeholder.